### ISSUE TITLE
PAM authentication unreliable

### ISSUE THREAD

**Issue by calz1** [jupyterhub/jupyterhub#1138]
This seems similar to #486 . 

I'm running this on CentOS 7:

> [data@localhost system]$ sudo pip3.6 show jupyterhub
> Name: jupyterhub
> Version: 0.7.2
> Summary: JupyterHub: A multi-user server for Jupyter notebooks
> Home-page: http://jupyter.org
> Author: Jupyter Development Team
> Author-email: jupyter@googlegroups.com
> 

I haven't opened the config file and am using the default PAM authentication while running jupyterhub as root via sudo. I can login fine once with my OS user called "data". When I login from another browser or another PC, I get invalid username or password in the web UI. I'm copying and pasting the username and password both times to eliminate typos as an issue. 

> 
> [I 2017-05-16 11:37:40.601 JupyterHub log:100] 302 POST /hub/login?next= (@::ffff:10.1.0.28) 3545.98ms
> [I 2017-05-16 11:37:40.601 JupyterHub login:91] User logged in: data
> [I 2017-05-16 11:37:40.660 JupyterHub log:100] 302 GET /hub/ (data@::ffff:10.1.0.28) 14.56ms
> [I 2017-05-16 11:37:40.679 data log:47] 302 GET /user/data (::ffff:10.1.0.28) 1.29ms
> [I 2017-05-16 11:37:40.783 JupyterHub log:100] 200 GET /hub/api/authorizations/cookie/jupyter-hub-token-data/[secret] (data@127.0.0.1) 30.40ms
> [I 2017-05-16 11:37:50.100 JupyterHub log:100] 302 GET / (@::ffff:10.1.0.28) 2.74ms
> [I 2017-05-16 11:37:50.137 JupyterHub log:100] 302 GET /hub (@::ffff:10.1.0.28) 0.46ms
> [I 2017-05-16 11:37:50.204 JupyterHub log:100] 302 GET /hub/ (@::ffff:10.1.0.28) 0.92ms
> [I 2017-05-16 11:37:50.246 JupyterHub log:100] 302 GET /login (@::ffff:10.1.0.28) 0.78ms
> [I 2017-05-16 11:37:50.332 JupyterHub log:100] 200 GET /hub/login (@::ffff:10.1.0.28) 50.68ms
> [W 2017-05-16 11:37:56.490 JupyterHub auth:471] PAM Authentication failed (data@::ffff:10.1.0.28): [PAM Error 7] Authentication failure
> [I 2017-05-16 11:37:56.504 JupyterHub log:100] 200 POST /hub/login?next= (@::ffff:10.1.0.28) 2184.06ms
>

**Comment by davidXire** [comment]
Hi, 

Same problem here with CentOS7 and pam 1.1.8... Any solution ? 
For information : I test on VM to upgrade PAM to 1.2.1 but I can't log into the system now...

**Comment by willingc** [comment]
@davidXire Do you see this error when you are using JupyterHub master? If you provide debug logs, that would help us better assist you. Thanks.

**Comment by davidXire** [comment]
I have the same trace like calz1

**Comment by willingc** [comment]
@davidXire Are you using JupyterHub master or 0.7.2? There's been a significant number of improvements since 0.7.2. Also, if you wish to provide your `jupyterhub_config.py` contents that would be helpful too.

**Comment by davidXire** [comment]
I have JupytherHub 0.7.2. I start jupyterhub with the command line "jupyterhub" so I used the default config.

**Comment by moizsajid** [comment]
I also have the same exact authentication problem for my regular(non-admin) user.

**Comment by calz1** [comment]
I was able to get the problem to go away by doing this in the config file, but it still appears to be a problem without it:

`c.PAMAuthenticator.open_sessions = False`

**Comment by davidXire** [comment]
Did you put this line in the "jupyterhub_config.py" config file ?

**Comment by calz1** [comment]
Correct.

**Comment by davidXire** [comment]
It's work, thanks a lot @calz1

**Comment by agustiin** [comment]
Thanks @calz1!

**Comment by jainsanmati** [comment]
Thanks a lot ! It worked @calz1

**Comment by ssword** [comment]
This issue still exists in Jupyterhub 0.9.4. But @calz1 's solution works like a charm.

**Comment by jbelsky** [comment]
Thanks @calz1 for this info.  I tried this but am still getting the same PAM authentication error:

```
[W 2018-10-25 13:44:41.747 JupyterHub auth:618] PAM Authentication failed (jasonbelsky): [PAM Error 7] Authentication failure
[W 2018-10-25 13:44:41.748 JupyterHub base:504] Failed login for jasonbelsky
```

I also set the ``c.PamAuthenticator.check_account = False`` just as a test in the ``jupyterhub_config.py``, but still got the same error.  I am running Jupyterhub 0.9.4 on CentOS Linux release 7.2.1511.  @vbraun from #486 , did you ultimately come up with any workaround?
